wasm-thumbnail/src/lib.rs

```diff
@@ -57,13 +57,36 @@ pub fn _resize_and_pad(
     let img = image::load_from_memory(slice)?;
 
     // Resize preserves aspect ratio
-    let img = img.resize(nwidth, nheight, FilterType::Lanczos3);
+    let img = img.resize(
+        std::cmp::min(nwidth, img.width()),
```

Comment:
Note: If I don't add this, images are frequently upscaled (poorly) and often overflow the max padded image size.